src/segment/filter/mod.rs

```diff
@@ -38,3 +45,56 @@ impl BloomConstructionPolicy {
         }
     }
 }
+
+enum FilterType {
+    StandardBloom = 0,
+    BlockedBloom = 1,
+}
+
+impl TryFrom<u8> for FilterType {
+    type Error = ();
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::StandardBloom),
+            1 => Ok(Self::BlockedBloom),
+            _ => Err(()),
+        }
+    }
+}
+
+pub trait AMQFilter: Sync + Send {
```

Comment:
Adding Sync + Send here, otherwise we would get errors from `blob_drop_after_flush`